keel-api/src/main/kotlin/com/netflix/spinnaker/keel/api/artifacts/DeliveryArtifact.kt

```diff
@@ -19,7 +19,18 @@ sealed class DeliveryArtifact {
   abstract val versioningStrategy: VersioningStrategy
   abstract val reference: String // friendly reference to use within a delivery config
   abstract val deliveryConfigName: String? // the delivery config this artifact is a part of
+
   override fun toString() = "$type:$name" + if (reference != null) " (ref: $reference)" else ""
+
+  override fun equals(other: Any?): Boolean {
+    if (other is DeliveryArtifact) {
+      return other.name == this.name &&
+        other.type == this.type &&
+        other.reference == this.reference &&
+        other.deliveryConfigName == this.deliveryConfigName
+    }
+    return super.equals(other)
```

Comment:
Why do this rather than just return false in the case where the other object isn't a DeliveryArtifact?

Reply:
Because I didn't want to handle other cases like `DeliveryArtifact?` (nullable type). This was my way of saying "other than the special case above, use the default behavior".

Reply:
@lorin I ended up changing tactics in this PR. Please take another look.